contracts/src/LightClient.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import { BN254 } from "bn254/BN254.sol";
+import { IPlonkVerifier } from "./interfaces/IPlonkVerifier.sol";
+import { IStakeTable } from "./interfaces/IStakeTable.sol";
+import { PlonkVerifier } from "./libraries/PlonkVerifier.sol";
+
+// TODO: replace these, for now WIP and test only
+import { VkTest } from "../test/stubs/Transfer1In2Out24DepthVk.sol";
+
+/// @notice A light client for HotShot consensus. Keeping track of its finalized states in safe,
+/// authenticated ways.
+contract LightClient {
+    // === Constants ===
+    //
+    /// @notice System parameter: number of blocks per epoch
+    uint32 public constant BLOCKS_PER_EPOCH = 20_000;
+
+    // === Storage ===
+    //
+    /// @notice genesis block commitment
+    LightClientState public genesisState;
+    /// @notice global storage of the finalized HotShot's light client state
+    LightClientState public finalizedState;
+    /// @notice current (finalized) epoch number
+    uint64 public currentEpoch;
+    /// @notice address of the stake table contract
+    IStakeTable internal stakeTable;
+
+    // === Data Structure ===
+    //
+    /// @notice The finalized HotShot state (as the digest of the entire HotShot state)
+    /// @param viewNum The latest view number of the finalized HotShot chain
+    /// @param blockHeight The block height of the latest finalized block
+    /// @param blockComm The block commitment (type: BN254::ScalarField)
```

Comment:
What is this? Latest block commitment or merkle commitment of all blocks? And why is it a field element as opposed to just a hash output?

Reply:
Ah, I see, because it's an input to the SNARK. However if in practice these commitments are generated from a hash function, they may not fit in a field element, we may need to hash to field.

Reply:
This is an error [due a misunderstanding from the bad variable name](https://github.com/EspressoSystems/HotShot/pull/2055#pullrequestreview-1737485954). It should be `block_comm_root` the root of the merkle tree that accumulates all the historical block commitments. (which should be using Rescue-based commitment, thus guaranteed to be a field element) 

I will update its name